common/src/main/java/com/redhat/cloud/notifications/models/SourcesSecretable.java

```diff
@@ -53,4 +53,12 @@ public interface SourcesSecretable {
      * @param basicAuthenticationSourcesId the ID of the secret.
      */
     void setBasicAuthenticationSourcesId(Long basicAuthenticationSourcesId);
+
+    Long getBearerAuthenticationSourcesId();
+
+    void setBearerAuthenticationSourcesId(Long bearerAuthenticationSourcesId);
+
+    String getBearerToken();
+
+    void setBearerToken(String bearerToken);
```

Comment:
Not for this PR, but I'm wondering if we should have an annotation (or similar) that takes care of adding the ID to the actual data.
```java
class MyProperties {
    @SourcesSecret
    String myTopSecretStuff;
}
```

Reply:
cc @MikelAlejoBR

Reply:
The `SecretUtils` class contains a few checks with the `SourcesSecretable` interface, in order to be able to store/update/delete those secrets from the service: https://github.com/RedHatInsights/notifications-backend/blob/master/common/src/main/java/com/redhat/cloud/notifications/routers/sources/SecretUtils.java#L92

I was thinking that we could totally use annotations for generating the getters/setters and storing the ID, but wouldn't that "break" the SecretUtils class? Or once generated, we should not have any problem?

Reply:
What do you mean?

Reply:
Maybe I didn't understand you right. I thought that with the `@SourcesSecret` annotation getters, setters and a property would be automatically generated for that entity.

The "problem" I see is that I don't know if it is possible to generate code that adheres to an interface, and make that entity conform to it?

I am thinking that I might have understood the whole thing wrong.

Reply:
One way would be that the annotation only marks an attribute as "This is secretable - expect a `theAttributeId` property with setter/getter.

`SecretUtils` would look for those annotations and continue as usual (I found the annotation, I know that there is a `setTheStuffId` and `getTheStuffId`) and i'll use them accordingly. 

The enhancement is that we wouldn't need a fixed interface and it's going to be more flexible with more secrets across different types.

More information, but I don't think this provide any value other than being somewhat interesting:

Talking about code generation (one alternative) is that we create a processor that fetches the `SourcesSecret` annotation and generates a new class `MyOriginalNameSecretEnhanced` with the `ids` (getter/setters). We would still need to do as above.

We should be able to use that class within the same code as if was something we wrote directly.

[This](https://github.com/josejulio/versioned-path) is something I build during the past hackathon that mimics that for something else. Ultimately I think this approach will be less complicated as less handling of the super class is required.

Reply:
All right, I have created a ticket so that we can apply this new approach: https://issues.redhat.com/browse/RHCLOUD-28488